exercises/high-scores/example.rs

```diff
@@ -0,0 +1,30 @@
+#[derive(Debug)]
+pub struct HighScores {
+    scores: Vec<u32>,
+}
+
+impl HighScores {
+    pub fn new(scores: &[u32]) -> Self {
+        Self {
+            scores: scores.to_vec(),
+        }
+    }
+
+    pub fn scores(&self) -> &[u32] {
+        &self.scores
+    }
+
+    pub fn latest(&self) -> u32 {
+        *self.scores.last().unwrap()
```

Comment:
We should not encourage students to use `unwrap` unconditionally like this. We should change the API to make the struct safer.

There are at least two possible ways to change this API to improve its safety:

- change the return type of `latest` and `personal_best` to `Option<u32>`
- change the return type of `new` to `Option<Self>`, returning None when the input list is empty

I do not have a preference between those two options. However, we should definitely add an appropriate test in which the input slice is empty to ensure that the code doesn't panic, as it would with `unwrap`.

Reply:
I think the first variant is a bit clearer. And the test is a good idea.